### PR TITLE
Process bundle batch in parallel 

### PR DESCRIPTION
**Pull request by frankjtao**

### CODE REVIEW

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseTransactionProcessor.java [hunk 1]
```diff
@@ -163,6 +171,16 @@ public void setVersionAdapter(ITransactionProcessorVersionAdapter theVersionAdap
...
 	public void start() {
 		ourLog.trace("Starting transaction processor");
+		myExecutor = new ThreadPoolTaskExecutor();
+		myExecutor.setThreadNamePrefix("bundle_batch_");
+		// For single thread set the value to 1
+		//myExecutor.setCorePoolSize(1);
+		//myExecutor.setMaxPoolSize(1);
+		myExecutor.setCorePoolSize(myDaoConfig.getBundleBatchPoolSize());
+		myExecutor.setMaxPoolSize(myDaoConfig.getBundleBatchMaxPoolSize());
+		myExecutor.setQueueCapacity(myDaoConfig.getBundleBatchQueueCapacity());
```
Comment (michaelabuckley) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseTransactionProcessor.java:181]: Do we need the complexity of a queue?  The main thread will wait for these to complete, so I wonder if we could delete this config.

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseTransactionProcessor.java [hunk 2]
```diff
@@ -309,59 +327,54 @@ public void setTxManager(PlatformTransactionManager theTxManager) {
...
+		IBase nextRequestEntry = null;
+		for (int i=0; i<requestEntriesSize; i++ ) {
+			nextRequestEntry = requestEntries.get(i);
+			BundleTask bundleTask = new BundleTask(completionLatch, theRequestDetails, responseMap, i, nextRequestEntry, theNestedMode);
+			myExecutor.submit(bundleTask);
+		}
 
-				myVersionAdapter.setResponseStatus(nextEntry, toStatusString(caughtEx.getException().getStatusCode()));
+		// waiting for all tasks to be completed
+		AsyncUtil.awaitLatchAndIgnoreInterrupt(completionLatch, 120L, TimeUnit.SECONDS);
```
Comment (michaelabuckley) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseTransactionProcessor.java:355]: Do we already have a request timeout configured somewhere?  This feels too short for some customers.

**Review by michaelabuckley — APPROVED** [review]
Looks good.

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseTransactionProcessor.java (hunk 1 above)
Comment (frankjtao) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseTransactionProcessor.java:181]: will delete this config

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseTransactionProcessor.java (hunk 2 above)
Comment (frankjtao) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseTransactionProcessor.java:355]: There is a connection timeout for 10000 milliseconds (10s). I can't find any other timeout value. The value here is 120s (2m). I can change it to 5m (300s).

**Pull request merged**